### ISSUE TITLE
Yolo Pixel Modification 

### ISSUE THREAD

**Issue by ronit450** [ultralytics/ultralytics#17533]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

one thing I have realized that pixel information of my image at any location such as row=91 col=207 is different before it goes to model and after it is loader and sent for training. There is no definite pattern there, so is there any way to check where did it changed my image information.

**Comment by Y-T-G** [comment]
It will be resized and normalized before being passed to the model, so it won't be the same as the original.

**Comment by ronit450** [comment]
@Y-T-G whre is the point it will be normalized and resized. I have kept self.augment = False

**Comment by Y-T-G** [comment]
Are you training or predicting?

**Comment by ronit450** [comment]
training

**Comment by Y-T-G** [comment]
`augment` argument doesn't control augmentations during training. It's not a training argument.

https://docs.ultralytics.com/modes/train/#train-settings

https://github.com/ultralytics/ultralytics/blob/4453ddab93c390c11c9f8bb8d90a761a4cec3342/ultralytics/models/yolo/detect/train.py#L59


https://github.com/ultralytics/ultralytics/blob/4453ddab93c390c11c9f8bb8d90a761a4cec3342/ultralytics/data/dataset.py#L181

**Comment by ronit450** [comment]
yes you are right, but the issue I had was,  I have 5 band image and training it to all 5 bands, but it completly changes the pixel information, like for testing I kept 4 bands active and last band as 0 and by checking the pixel information it gave [ 44 0 125 154 0], whereas in Yolo before it goes for training pass, it gave [ 83 41 17 0 5] , which does not make sense at all. I just want to check a point where it transforms image pixels

**Comment by glenn-jocher** [comment]
@ronit450 the pixel transformation occurs during the data loading and preprocessing stage in `ultralytics/data/dataset.py`. The image is converted to float32, normalized to 0-1 range, and resized during this process. If you need to preserve exact pixel values, you'll need to modify the preprocessing pipeline in the Dataset class.

For debugging, you can add print statements in `dataset.py` to track the pixel values at each transformation step.
